include/libvfio-user.h

```diff
@@ -617,6 +617,18 @@ vfu_get_pci_non_std_config_space(vfu_ctx_t *vfu_ctx);
 int
 vfu_ctx_try_attach(vfu_ctx_t *vfu_ctx);
 
+/*
+ * Finalizes the device making it ready for vfu_ctx_drive or vfu_ctx_try_attach.
+ * This function is optional as it is automatically called by vfu_ctx_drive or
+ * vfu_ctx_try_attach. Calling it multiple times is idempotent.
+ *
+ * @vfu_ctx: the libvfio-user context
+ *
+ * @returns: 0 on success, -1 on error. Sets errno.
+ */
+int
+vfu_realize_ctx(vfu_ctx_t *vfu_ctx);
```

Comment:
I do not quite understand the value of making vfu_realize_ctx() API public. Anyways this is optional and ctx realisation is taken care automatically internally.

Reply:
It's nice to separate out the part that accept()s a client from ongoing operations in the later drive().

It would be nice if this documented the blocking for a client behaviour here though.